From 3670d645f4a26a0a9e87e7f3a8608e7cc1d53b5b Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Wed, 12 Aug 2020 14:04:17 -0600
Subject: [PATCH] Don't use Xlib headers by default.

Conceptually, defining EGLNativeDisplayType to Display* is simply incorrect,
because it's not always going to be an Xlib display.

In practice, more and more programs and libraries are being built with EGL but
without Xlib, and end up breaking when the Xlib headers are missing.

Rather than requiring every such program to define EGL_NO_X11, use the (void*)
typedef by default. The existing USE_X11 header can still preserve the current
behavior.

The change to EGLNativeDisplayType will be fine for eglGetDisplay, since you
can always pass a (Display *) as a (void *) parameter.

The EGLNativePixmapType and EGLNativeWindowType types don't change, because
khronos_uintptr_t and XID are both typedefs to unsigned long.

The only things that might break are calls to eglQueryNativeDisplayNV, or
programs that do need the Xlib headers and expect egl.h to include them. Both
of those cases would be trivial to fix.
---
 api/EGL/eglplatform.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/api/EGL/eglplatform.h b/api/EGL/eglplatform.h
index 1edbafa..99362a2 100644
--- a/include/EGL/eglplatform.h
+++ b/include/EGL/eglplatform.h
@@ -103,13 +103,7 @@ typedef intptr_t EGLNativeDisplayType;
 typedef intptr_t EGLNativePixmapType;
 typedef intptr_t EGLNativeWindowType;
 
-#elif defined(__unix__) && defined(EGL_NO_X11)
-
-typedef void             *EGLNativeDisplayType;
-typedef khronos_uintptr_t EGLNativePixmapType;
-typedef khronos_uintptr_t EGLNativeWindowType;
-
-#elif defined(__unix__) || defined(USE_X11)
+#elif defined(USE_X11)
 
 /* X11 (tentative)  */
 #include <X11/Xlib.h>
@@ -119,6 +113,12 @@ typedef Display *EGLNativeDisplayType;
 typedef Pixmap   EGLNativePixmapType;
 typedef Window   EGLNativeWindowType;
 
+#elif defined(__unix__)
+
+typedef void             *EGLNativeDisplayType;
+typedef khronos_uintptr_t EGLNativePixmapType;
+typedef khronos_uintptr_t EGLNativeWindowType;
+
 #elif defined(__APPLE__)
 
 typedef int   EGLNativeDisplayType;
